src/client/auth.rs

```diff
@@ -279,7 +286,12 @@ impl AuthMechanism {
                 x509::build_speculative_client_first(credential),
             )))),
             Self::Plain => Ok(None),
```

Comment:
This is a little troubling here. It's necessary because the cache is using a tokio::sync::RwLock. Would it be better to make build_speculative_client async? I'm not sure on the repercussions of that. This is probably safe enough?